modules/phase_field/include/kernels/MultiGrainRigidBodyMotion.h

```diff
@@ -0,0 +1,30 @@
+/****************************************************************/
+/* MOOSE - Multiphysics Object Oriented Simulation Environment  */
+/*                                                              */
+/*          All contents are licensed under LGPL V2.1           */
+/*             See LICENSE for full restrictions                */
+/****************************************************************/
+#ifndef MULTIGRAINRIGIDBODYMOTION_H
+#define MULTIGRAINRIGIDBODYMOTION_H
+
+#include "GrainRigidBodyMotionBase.h"
+
+//Forward Declarations
+class MultiGrainRigidBodyMotion;
+
+template<>
+InputParameters validParams<MultiGrainRigidBodyMotion>();
+
+class MultiGrainRigidBodyMotion : public GrainRigidBodyMotionBase
+{
+public:
+  MultiGrainRigidBodyMotion(const InputParameters & parameters);
+
+protected:
+  virtual Real computeQpResidual();
+  virtual Real computeQpJacobian();
+  virtual Real computeQpOffDiagJacobian(unsigned int jvar);
+  virtual Real computeQpCJacobian();
```

Comment:
I'm not a huge fan of this function name (which seems to have been originally introduced by @tonkmr in his `SoretDiffusion` class and emulated here) -- at first I thought it was some kind of typo...  I'd also like to know a bit more about why this needs to exist outside of the usual `computeQpOffDiagJacobian()` hierarchy?  If `jvar==_c_var`, why not just call `Base::computeQpOffDiagJacobian()`?